server/parsec/components/postgresql/invite.py

```diff
@@ -1063,6 +1063,20 @@ async def test_dump_all_invitations(
     # New invite transport API
     # TODO: Remove the old API once the new one is fully functional
 
+    def is_greeter_allowed(
+        self,
+        invitation_type: InvitationType,
+        invitation_created_by: UserID,
+        greeter_id: UserID,
+        greeter_profile: UserProfile,
+    ) -> bool:
+        if invitation_type == InvitationType.DEVICE:
+            return invitation_created_by == greeter_id
+        elif invitation_type == InvitationType.USER:
+            return greeter_profile == UserProfile.ADMIN
+        else:
+            raise NotImplementedError
```

Comment:
This logic is also implemented in `components/memory/invite.py`:
https://github.com/Scille/parsec-cloud/blob/9ed86452345a54058ac6f17f5b5d266d7bac01f6/server/parsec/components/memory/invite.py#L658-L664

Can it be moved up to `components/invite.py`? to share it between `memory` and `postgresql` implementations?